### PR TITLE
[Bullet] Collision Features

### PR DESCRIPTION
**Pull request by Lobotuerk**
Collision seems to be working, deleting and creating does not segfault, bodies seem to move properly

### CODE REVIEW

File: bullet/src/EntityManagementFeatures.cc [hunk 1]
```diff
@@ -36,22 +36,39 @@ bool EntityManagementFeatures::RemoveModel(const Identity &_modelID)
...
       this->worlds.at(model->world)->world->removeRigidBody(linkInfo->link);
+      delete linkInfo->link;
       it = this->links.erase(it);
       continue;
     }
     it++;
   }
 
+
+
```
Comment (Blast545) [inline bullet/src/EntityManagementFeatures.cc:71]:
```suggestion
```

File: bullet/src/EntityManagementFeatures.cc [hunk 2]
```diff
@@ -36,22 +36,39 @@ bool EntityManagementFeatures::RemoveModel(const Identity &_modelID)
   }
   // Current implementation does not include collisions nor joints
```
Comment (Blast545) [inline bullet/src/EntityManagementFeatures.cc:37]:
```suggestion
  // Current implementation does not include joints
```

File: bullet/src/SimulationFeatures.hh [hunk 3]
```diff
@@ -40,6 +42,11 @@ class SimulationFeatures :
       ForwardStep::Output &_h,
       ForwardStep::State &_x,
       const ForwardStep::Input &_u) override;
+
+  public: std::vector<ContactInternal> GetContactsFromLastStep(
+      const Identity &/* _worldID */) const override
+      { // TODO(lobotuerk): Implement contacts getter, could be like https://pybullet.org/Bullet/phpBB3/viewtopic.php?t=2855
```
Comment (Blast545) [inline bullet/src/SimulationFeatures.hh:48]: If this is a Dummy implementation, I suggest printing to ignwrn something to be aware when it's used JIC

File: bullet/src/SDFFeatures.cc [hunk 4]
```diff
@@ -69,23 +75,101 @@ Identity SDFFeatures::ConstructSdfLink(
...
+  //                             ->Get<btScalar>("restitution_coefficient");
+  if (shape != nullptr)
+  {
+    const auto &linkInfo = this->links.at(_linkID);
+    const auto &modelID = linkInfo->model;
+    const auto &modelInfo = this->models.at(modelID);
+    const auto &link = linkInfo->link;
+    const auto &world = this->worlds.at(modelInfo->world)->world;
+    // btTransform transform;
+    delete link->getCollisionShape();
```
Comment (Blast545) [inline bullet/src/SDFFeatures.cc:162]: Please leave a comment here stating why it's solved this way here. Maybe that can change in the future

**Review by Blast545 — APPROVED** [review]
Minor comments, otherwise LGTM

File: bullet/src/SimulationFeatures.hh (hunk 3 above)
Comment (Lobotuerk) [reply bullet/src/SimulationFeatures.hh]: Updated on 30ae58a

File: bullet/src/SDFFeatures.cc (hunk 4 above)
Comment (Lobotuerk) [reply bullet/src/SDFFeatures.cc:162]: Updated on 30ae58a, hopefully, this does not spam the console

**Comment by Lobotuerk** [comment]
Merging after addresing everything

**Pull request merged**